docs/rule/no-invalid-interactive.md

```diff
@@ -5,6 +5,12 @@
 Adding interactivity to an element that is not naturally interactive content leads to a very poor experience for
 users of assistive technology (i.e. screen readers). In order to ensure that screen readers can provide useful information to their users, we should add an appropriate `role` attribute when the underlying element would not have made that role obvious.
 
+Some complexities of note:
+
+Both a `tabindex` attribute and an interactive role is required to make a custom element interactive; otherwise it is a non-interactive element and not eligible to have an interaction added.
+
+The `<canvas>` element is not an interactive element, but may contain certain interactive elements as descendants. From the HTML spec: `a` elements, `img` elements with `usemap` attributes, `button` elements, `input` elements whose `type` attribute are in the Checkbox or Radio Button states, `input` elements that are buttons, and `select` elements with a multiple attribute or a display size greater than 1.
```

Comment:
The "From the HTML spec" bit should link to the section for each of these elements for easier reading / verification.

Reply:
> The `<canvas>` element is not an interactive element, but may contain certain interactive elements as descendants.

Isn't this true of _basically_ every non-interactive element that is allowed to contain any content? Why is `canvas` called out specifically?

Reply:
> `input` elements whose `type` attribute are in the Checkbox or Radio Button states

The spec specifically says that any `input` that is **not** `type="hidden"` is interactive.

Reply:
Still open issues here:

- [ ] Change prose RE: `input`; all non-`type="hidden"` inputs are interactive we shouldn't imply otherwise
- [ ] Need to link to the spec in the various spots